query/security/wcpm/src/test/groovy/org/codice/ddf/admin/security/wcpm/actions/persist/SaveContextPoliciesTest.groovy

```diff
@@ -0,0 +1,254 @@
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role"
+        ]
+
+        testData = [
+                policies: [
+                        [
+                                paths        : [ '/' ],
+                                authTypes    : [ 'basic' ],
+                                realm        : 'karaf',
+                                claimsMapping: [
+                                        [
+                                                claim     : 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role',
+                                                claimValue: 'system-admin'
+                                        ]
+                                ]
+                        ],
+                        [
+                                paths        : [ '/test' ],
+                                authTypes    : [ 'SAML', 'PKI' ],
+                                realm        : 'karaf',
+                                claimsMapping: [
+                                        [
+                                                claim     : 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role',
+                                                claimValue: 'system-admin'
+                                        ]
+                                ]
+                        ]
+                ]
+        ]
+
+        operationReport = Mock(OperationReport)
+        configuratorFactory = Mock(ConfiguratorFactory)
+        configurator = Mock(Configurator)
+        configReader = Mock(ConfigReader)
+        configurator.updateConfigFile({ it == PolicyManagerServiceProperties.POLICY_MANAGER_PID }, _, _) >> {
+            args -> policyManager.setPolicies(args[1])
+        }
+
+        stsConfig = [(StsServiceProperties.STS_CLAIMS_PROPS_KEY_CLAIMS): testClaims ]
+        configReader.getConfig(_) >> stsConfig
+
+        policyManager = new PolicyManager()
+        ListField<ContextPolicyBin> contextPolicies = new ListFieldImpl<>(ContextPolicyBin.class)
+        contextPolicies.setValue(testData.policies)
+        policyManager.setPolicies(new PolicyManagerServiceProperties().contextPoliciesToPolicyManagerProps(contextPolicies.getList()))
+
+        configurator.commit(_,_) >> operationReport
+        configReader.getServiceReference(_) >> policyManager
+        configuratorFactory.getConfigurator() >> configurator
+        configuratorFactory.getConfigReader() >> configReader
+        actionCreator = new WcpmActionCreator(configuratorFactory)
+        action = actionCreator.createAction(SaveContextPolices.ACTION_ID)
+    }
+
+    def 'Pass with valid update' () {
+        setup:
+        operationReport.containsFailedResults() >> false
+
+        when:
+        action.setArguments(testData)
+        ActionReport report = action.process()
+
+        then:
+        report.messages().isEmpty()
+    }
+
+    def 'Fail when failed to persist' () {
```

Comment:
Ensure all tests are returning null in the report if the action fails to process